apps/studio/src/pages/sites/[siteId]/admin.tsx

```diff
@@ -0,0 +1,228 @@
+    id: siteId,
+  })
+  const [previousNavbar] = trpc.site.getNavbar.useSuspenseQuery({
+    id: siteId,
+  })
+  const [previousFooter] = trpc.site.getFooter.useSuspenseQuery({
+    id: siteId,
+  })
+
+  // NOTE: Refining the setNotificationSchema here instead of in site.ts since omit does not work after refine
+  const {
+    register,
+    handleSubmit,
+    formState: { isDirty, errors },
+  } = useZodForm({
+    schema: setSiteConfigAdminSchema
+      .omit({ siteId: true })
+      .refine((data) => !!data.config, {
+        message: "Site config must be present",
+        path: ["config"],
+      })
+      .refine((data) => !!data.theme, {
+        message: "Site theme must be present",
+        path: ["theme"],
+      })
+      .refine((data) => !!data.navbar, {
+        message: "Site navbar must be present",
+        path: ["navbar"],
+      })
+      .refine((data) => !!data.footer, {
+        message: "Site footer must be present",
+        path: ["footer"],
+      }),
+    defaultValues: {
+      config: JSON.stringify(previousConfig, null, 2),
+      theme: JSON.stringify(previousTheme, null, 2),
+      navbar: JSON.stringify(previousNavbar.content, null, 2),
+      footer: JSON.stringify(previousFooter.content, null, 2),
+    },
+  })
+
+  const [nextUrl, setNextUrl] = useState("")
+  const isOpen = !!nextUrl
+
+  useEffect(() => {
+    const handleRouteChange = (url: string) => {
+      if (isDirty) {
+        router.events.off("routeChangeStart", handleRouteChange)
+        setNextUrl(url)
+        router.events.emit("routeChangeError")
+        // eslint-disable-next-line @typescript-eslint/only-throw-error
+        throw "Error to abort router route change. Ignore this!"
+      }
+    }
+
+    if (!isOpen) {
+      router.events.on("routeChangeStart", handleRouteChange)
+    }
+    return () => {
+      router.events.off("routeChangeStart", handleRouteChange)
+    }
+  }, [isOpen, router.events, isDirty])
+
+  const onClickUpdate = handleSubmit((input) => {
+    mutate({
+      siteId,
+      ...input,
+    })
+  })
+
+  if (!isUserIsomerAdmin) {
+    return (
+      <Center h="full">
+        <Text>You do not have permission to access this page.</Text>
+      </Center>
+    )
+  }
+
+  return (
+    <>
+      <UnsavedSettingModal
+        isOpen={isOpen}
+        onClose={() => setNextUrl("")}
+        nextUrl={nextUrl}
+      />
+      <chakra.form
+        onSubmit={onClickUpdate}
+        overflow="auto"
+        height={0}
+        minH="100%"
+      >
+        <Center py="5.5rem" px="2rem">
+          <VStack w="48rem" alignItems="flex-start" spacing="1.5rem">
+            <Text w="full" textStyle="h3-semibold">
+              Manage site configurations
+            </Text>
+            <Infobox variant="warning" textStyle="body-2" size="sm">
+              No validation is done on the JSON input. Please ensure that they
+              are valid before saving.
```

Comment:
can we warn the migrators that these changes are **immediately** published? since this will be on prod, it might not be very evident to the migrators but it'l impact the site hugely (eg: we add a link on footer before the page (which is linked) is created)

Reply:
Yup the warning is shown at the bottom next to the save button, same design as the site settings page.